convert.go

```
@@ -0,0 +1,92 @@
+// eventTypeAdapter preserves all original event properties while converting data type
+type eventTypeAdapter[T, M any] struct {
+	original Event[T]
+	data     M
+	props    map[string]any // stores converted properties
+}
+
+func (a *eventTypeAdapter[T, M]) Name() string { return a.original.Name() }
+func (a *eventTypeAdapter[T, M]) Data() M      { return a.data }
+func (a *eventTypeAdapter[T, M]) SetData(d M) (Event[M], error) {
+	a.data = d
+	if setter, ok := a.original.(interface{ SetData(T) (Event[T], error) }); ok {
```

golangci-lint (1.64.8)

SA5010: impossible type assertion; Event[T] and interface{SetData(T) (Event[T], error)} contradict each other:
	wrong type for SetData method
		have func(T) (go.lumeweb.com/event.Event[T], error)
		want func(T) (go.lumeweb.com/event.Event[T], error)

(staticcheck)